models/repo/repo.go

```diff
@@ -217,6 +217,20 @@ func (repo *Repository) IsBroken() bool {
 	return repo.Status == RepositoryBroken
 }
 
+// IsPinned indicates that repository is pinned
+func (repo *Repository) IsPinned() bool {
```

Comment:
I think the function should be `IsPinned(userID int64)`. One could pin another public repository?

Reply:
A repo is either pinned to its owner's profile, or not at all.

This is maybe the same confusion I had with @delvh ? This is not an implementation for pinning any repo you have access to to your profile - it's an implementation for pinning a repo to its *owner's* profile.

Reply:
Hm, the most interesting thing I happen to work on (and would find worth a pin to my user profile) is in an org I own, not in my user namespace. So I couldn't pin that then? I'd assume that's a common use case, but maybe that's just me.

Reply:
It sounds like that's everyone's else's view too. This may need reimplementing with that in mind.